acceptance/framework/consul/helm_cluster.go

```diff
@@ -470,3 +452,25 @@ func defaultValues() map[string]string {
 	}
 	return values
 }
+
+func CreateOrUpdateK8sSecret(t *testing.T, client kubernetes.Interface, cfg *config.TestConfig, namespace string, secretName string, secretKey string, secret string) {
+	_, err := client.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, err := client.CoreV1().Secrets(namespace).Create(context.Background(), &corev1.Secret{
```

Comment:
Is there value in passing in `context` and using that here?

Reply:
Great question!

I don't think at the moment, personally.  For me, dependency injection is valuable when:
- you have multiple types of an interface and you want to exchange them (we don't here)
- you want to pass in a mock for testing purposes (a big topic that I'll hint at below)

Code can always be refactored so if these two needs change, I would prefer to inject it then.

We try to push for testing workflows with little mocking and stubbing and so DI becomes less of a concern.  Some people prefer to write all of their code this way and at one point I did, but I found that unless one of the two above things were true (with the first item being more important) creating DI patterns provided less value...given that I could easily refactor when needed.

When we also look at this file and context is not injected anywhere, would I just change only this function? Would I modify all the other references to context in this file and possibly their callers? If I am injecting this are there other things I should then inject?  I think it just muddies up the code change and the conversation and I am brought back to the `why?` and the two points I mentioned above.

Thoughts?

Reply:
Excellent reasoning! I totally agree. I just saw the opportunity and wanted to ask if you had considered it. I think your argument totally makes sense.